File: Walter/LENGUAJE_ESTRUCTURADO_12062023-main/APACHENETBEANS/src/herencia_figurageometrica2d/Rectangulo.java
package herencia_figurageometrica2d;

public class Rectangulo extends FiguraGeometrica2D {
    
    private double largo;
    private double ancho;

    public Rectangulo() {
    }

    public Rectangulo(double largo, double ancho) {
        this.largo = largo;
        this.ancho = ancho;
    }

    public double getLargo() {
        return largo;
    }

    public void setLargo(double largo) {
        this.largo = largo;
    }

    public double getAncho() {
        return ancho;
    }

    public void setAncho(double ancho) {
        this.ancho = ancho;
    }

    @Override
    public String toString() {
        return "Rectangulo{" + "largo=" + largo + ", ancho=" + ancho + '}';
    }

    @Override
    public double area() {
        return this.largo * this.ancho;
    }

    @Override
    public double perimetro() {
        return 2 * this.largo + 2 * this.ancho;
    }

    @Override
    public String soy() {
        return "Rectangulo";
    }
    
    
}

File: Walter/Python_walter/POO/Ejemplo6/rectangulo.py
from figurageometrica import FiguraGeometrica

class Rectangulo(FiguraGeometrica):

      def __init__(self, largo=None, ancho=None):
          self.largo = largo
          self.ancho = ancho

      def perimetro(self):
          return 2 * self.largo + 2 * self.ancho
      
      def area(self):
          return self.largo * self.ancho
      
      def soy(self):
          return 'Rectangulo'